lib/rack/encryptor.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'openssl'
+require 'securerandom'
+require 'zlib'
+
+module Rack
+  class Encryptor
+    class Error < StandardError
+    end
+
+    class InvalidSignature < Error
+    end
+
+    class InvalidMessage < Error
+    end
+
+    DEFLATE_BIT = 0x1000
+
+    # The secret String must be at least 64 bytes in size. The first 32 bytes
+    # will be used for the encryption cipher key. The remainder will be used
+    # for an HMAC key.
+    #
+    # Options may include:
+    # * :serialize_json
+    #     Use JSON for message serialization instead of Marshal. This can be
+    #     viewed as a security ehancement.
+    # * :gzip_over
+    #     For message data over this many bytes, compress it with the deflate
+    #     algorithm.
+    # * :purpose
+    #     Limit messages to a specific purpose. This can be viewed as a
+    #     security enhancement to prevent message reuse from different contexts
+    #     if keys are reused.
```

Comment:
Roda uses the cookie key for this.  Do you think there is a reason to use a separate option?  If so, do you think using the cookie key is a good default?

Reply:
This feature is similar to an option offered by `ActiveSupport::MessageEncryptor`: https://api.rubyonrails.org/v5.2.3/classes/ActiveSupport/MessageEncryptor.html#class-ActiveSupport::MessageEncryptor-label-Confining+messages+to+a+specific+purpose

My hope was keep `Rack::Encryptor` as general purpose as possible, hence adding this option. It's used in the same way Roda uses the cookie key name as part of the authenticated message. Generally, I think this is good idea as it does add some effective security. However, one should not be reusing keys in the first place, but you never know!

Reply:
I'm fine keeping the option.  However, unless you have an objection to it, I would default the value to the cookie key.

Reply:
I now realized I didn't wire up using the `:purpose` option from session/cookie.rb. Will add that and will use the cookie name as the value.